src/align.py

```diff
@@ -16,8 +19,24 @@ def get_edits(p: str, q: str) -> tuple[str, str, str]:
 
     """
     assert len(p) == len(q)
-    # FIXME: do the actual calculations here
-    return '', '', ''
+    M = ''
+    i = 0
+    j = 0
+    while (i<len(p)) and (j<len(q)) :
+        if(p[i]== '-'):
+            p = p[:i] + p[i+1:]
```

Comment:
You are copying `len(p) - 1` elements here. And you are inside a loop that runs `len(p)` times. This has the makings of an $O(n^2)$ algorithm...